Source/UI/Cards/RoomCard.swift

```diff
@@ -49,9 +49,10 @@ struct RoomCard: View {
                                 .foregroundColor(.secondaryText)
                         }
                         .onChange(of: viewModel.alias) { newValue in
+                            viewModel.errorMessage = nil
                             viewModel.alias = newValue
                                 .lowercased()
-                                .filter("abcdefghijklmnopqrstuvwxyz0123456789-".contains)
+                                .filter("abcdefghijklmnopqrstuvwxyz0123456789".contains)
```

Comment:
This works, however according to the spec dashes should be valid in an alias. The rooms 2.0 spec points to [RFC 1035](https://www.rfc-editor.org/rfc/rfc1035) which says 
> Note that while upper and lower case letters are allowed in domain
names, no significance is attached to the case.  That is, two names with
the same spelling but different case are to be treated as if identical.

> The labels must follow the rules for ARPANET host names.  They must
start with a letter, end with a letter or digit, and have as interior
characters only letters, digits, and hyphen.  There are also some
restrictions on the length.  Labels must be 63 characters or less.

So maybe the server side is at fault here? This doesn't feel like a blocker to merging this PR, but I think we should open an issue for it if you don't want to spend more time on it here.

Reply:
>So maybe the server side is at fault here? 

Submitting an alias with hyphens anywhere in the string would cause a server side error.

Reply:
Additionally, I was allowed to create an alias starting with a digit.

<img width="348" alt="image" src="https://user-images.githubusercontent.com/1332769/204810782-e4d91392-375b-4b7a-ada1-98d38b95bc26.png">